code/__defines/_tick.dm

```diff
@@ -1,13 +1,27 @@
-#define TICK_LIMIT_RUNNING 80
-#define TICK_LIMIT_TO_RUN 78
+/// Percentage of tick to leave for master controller to run
+#define MAPTICK_MC_MIN_RESERVE 70
+#define MAPTICK_LAST_INTERNAL_TICK_USAGE (world.map_cpu)
+
+/// Tick limit while running normally
+#define TICK_BYOND_RESERVE 2
+#define TICK_LIMIT_RUNNING (max(100 - TICK_BYOND_RESERVE - MAPTICK_LAST_INTERNAL_TICK_USAGE, MAPTICK_MC_MIN_RESERVE))
+/// Tick limit used to resume things in stoplag
+#define TICK_LIMIT_TO_RUN 70
+/// Tick limit for MC while running
 #define TICK_LIMIT_MC 70
-#define TICK_LIMIT_MC_INIT_DEFAULT 98
+/// Tick limit while initializing
+#define TICK_LIMIT_MC_INIT_DEFAULT (100 - TICK_BYOND_RESERVE)
+
 
-#define TICK_USAGE world.tick_usage //for general usage
-#define TICK_USAGE_REAL world.tick_usage    //to be used where the result isn't checked
+/// for general usage of tick_usage
+#define TICK_USAGE world.tick_usage
+/// to be used where the result isn't checked
+#define TICK_USAGE_REAL world.tick_usage
```

Comment:
What is the purpose of having two defines with identical values?

Reply:
I don't know. All modern codebases (tg, nebula, bay) have two identical defines. TICK_USAGE_REAL used less and I found only two places where it used. TICK_USAGE used more often tho.

Reply:
If you want to keep them distinct, clarify the comment as to their expected usage. `/// to be used where the result isn't checked` is nonsensical to me: result of what? It returns a number; obviously you're checking it, not just calling it for the sake of calling it.